File: Problems/Basic/SortString/sortString.js
function sortString(str) {
  console.log("Unsorted string:", str);

  // Initialize frequency array for 26 lowercase English letters
  const freq = new Array(26).fill(0);

  // Count the frequency of each character
  for (let i = 0; i < str.length; i++) {
    freq[str.charCodeAt(i) - "a".charCodeAt(0)]++;
  }

  // Build the sorted string
  let sorted = "";
  for (let i = 0; i < 26; i++) {
    while (freq[i]-- > 0) {
      sorted += String.fromCharCode(i + "a".charCodeAt(0));
    }
  }

  console.log("Sorted string:", sorted);
  return sorted;
}

// Example usage
const input = "worldhello";
sortString(input);

File: Problems/Basic/SortString/sortString.cpp
/*
 * Write a program to sort the string
 * Assume string will always be in lowercase
 */

#include <bits/stdc++.h>
using namespace std;

int main()
{
    string str = "worldhello";
    cout << "Unsorted string: " << str << endl;
    int freq[26] = {0};

    // Count the frequency of each character
    for (char c : str)
    {
        freq[c - 'a']++;
    }

    // Build the sorted string
    string sorted = "";
    for (int i = 0; i < 26; i++)
    {
        while (freq[i]-- > 0) // freq[i] minus minus i.e decrement
        {
            sorted += i + 'a';
        }
    }

    cout << "Sorted string: " << sorted << endl;

    return 0;
}